lib/components/ModalPortal.js

```diff
@@ -30,8 +22,22 @@ export default class ModalPortal extends Component {
     closeTimeoutMS: PropTypes.number,
     shouldCloseOnOverlayClick: PropTypes.bool,
     onRequestClose: PropTypes.func,
-    className: PropTypes.string,
-    overlayClassName: PropTypes.string,
+    className: PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.shape({
+        base: PropTypes.string.isRequired,
+        afterOpen: PropTypes.string.isRequired,
+        beforeClose: PropTypes.string.isRequired
+      })
+    ]),
+    overlayClassName: PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.shape({
+        base: PropTypes.string.isRequired,
+        afterOpen: PropTypes.string.isRequired,
+        beforeClose: PropTypes.string.isRequired
+      })
```

Comment:
The suffixes are not flexible. There are BEM [alternative naming schemes](https://en.bem.info/methodology/naming-convention/#alternative-naming-schemes). The suffixes make impossible of using them. Also, we can pass props like this (without BEM example):

```javascript
{
  base: 'modal',
  afterOpen: 'open',
  beforeClose: 'close'
}
```